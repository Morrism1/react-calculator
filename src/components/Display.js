import PropTypes from 'prop-types';

export default function Display({ result }) {
  return (
    <thead>
      <tr>
        <th colSpan={5}>
          <div className="calc-display" id="display">
            {result}
          </div>
        </th>
      </tr>
    </thead>

  );
}

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: 0,
};
