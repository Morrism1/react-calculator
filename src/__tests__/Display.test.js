import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../components/Display';
import '@testing-library/jest-dom/extend-expect';

test('Render the display component  correctly', () => {
  const displayTree = renderer.create(<Display />).toJSON();
  expect(displayTree).toMatchSnapshot();
});
