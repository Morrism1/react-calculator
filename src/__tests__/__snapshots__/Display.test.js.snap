// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the display component  correctly 1`] = `
<thead>
  <tr>
    <th
      colSpan={5}
    >
      <div
        className="calc-display"
        id="display"
      />
    </th>
  </tr>
</thead>
`;
